src/Features/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.Features.csproj

```diff
@@ -72,6 +72,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Humanizer.Core" Version="$(HumanizerCoreVersion)" PrivateAssets="compile" />
+    <PackageReference Include="ICSharpCode.Decompiler" Version="$(ICSharpCodeDecompilerVersion)" />
```

Comment:
cc @jaredpar to advise if this will be a problem for source build

Reply:
Guessing this will be a problem but checking.

Reply:
Yes, I believe this would be a problem. Features need to be source-buildable.

Reply:
You can add the reference to LanguageServer and MEF export it there, no?

Reply:
@tmat Can't add an editor features reference to the language server project.  I was thinking of moving it to the Microsoft.CodeAnalysis.LanguageServer.Protocol.csproj, but it requires dependencies on C# types.  While [we do technically depend](https://github.com/dotnet/roslyn/blob/main/src/Features/LanguageServer/Protocol/Microsoft.CodeAnalysis.LanguageServer.Protocol.csproj#L15) on it there, it really isn't supposed to have language specific dependencies.  Same issue occurs if I wanted to link it to the Microsoft.CodeAnalysis.LanguageServer project.  I can technically do it, but violates language agnostic terms

Reply:
Two options:
1) Make this service language agnostic
2) Add Microsoft.CodeAnalysis.LanguageServer.CSharp

I'd try [1] first. `CSharpDecompiledSourceFormattingRule` can be moved to Features.CSharp.

Reply:
By "this service" I mean the actual decompilation. Could be a new service.
The glue around it, which might be C# specific could be moved to Features.CSharp

Reply:
Updated - moved the CSharp bits to CSharp.Features and added an abstraction over the decompilation, moved the actual decompilation implementation to LanguageServer.Protocol